src/components/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,124 @@
+import PopUp from './PopUp'
+import { Button, ModalProps, Spinner } from 'react-bootstrap'
+import { useForm } from 'react-hook-form'
+import Input from './Input'
+import { useEffect } from 'react'
+
+interface IChangePasswordModal extends ModalProps {}
+
+export interface IChangePasswordModalForm {
+  password: string
+  newPassword: string
+  newPasswordConfirm: string
+}
+
+export default function ChangePasswordModal({
+  onSubmit,
+  onHide,
+  ...modalProps
+}: IChangePasswordModal) {
+  const {
+    register,
+    watch,
+    getFieldState,
+    handleSubmit,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm({
+    mode: 'onSubmit',
+    defaultValues: {
+      password: '',
+      newPassword: '',
+      newPasswordConfirm: '',
+    },
+  })
+
+  useEffect(() => {
+    !modalProps.show && reset()
+  }, [modalProps.show, reset])
+
+  const handleClose = () => {
+    onHide && onHide()
+    reset()
+  }
+
+  function validationProps(fieldName: keyof IChangePasswordModalForm) {
+    const { error } = getFieldState(fieldName)
+    return {
+      isInvalid: !!error?.message,
+      error: error?.message,
+    }
+  }
+
+  return (
+    <PopUp
+      closeButton
+      title="Change Password"
+      {...modalProps}
+      onHide={handleClose}
+    >
+      <div>
+        <p>Please enter a new password of at least 8 characters.</p>
+      </div>
+      <form onSubmit={handleSubmit(onSubmit)} className="vertical-rhythm px-2">
```

Comment:
No need for the px-2.
```suggestion
      <form onSubmit={handleSubmit(onSubmit)} className="vertical-rhythm">
```